### DISCUSSION TITLE
Error: ENOENT: no such file or directory, scandir 'C:\Users\user\Desktop\nextjs-blog\posts'

### DISCUSSION THREAD

**Discussion by mohsenghalandar** [vercel/next.js#36347] [Help]
I did every thing as instruction - make lib/post.js in upper directory and put some code from NEXT.js website and  copy them to this 
`import fs from 'fs'`
`import path from 'path'`
`import matter from 'gray-matter'`

`const postsDirectory = path.join(process.cwd(), 'posts')`

`export function getSortedPostsData() {`
 ` // Get file names under /posts
  ``const fileNames = fs.readdirSync(postsDirectory)
  `const allPostsData = fileNames.map(fileName => {
   ` // Remove ".md" from file name to get id
   `` const id = fileName.replace(/\.md$/, '')`

    `// Read markdown file as string
   ` const fullPath = path.join(postsDirectory, fileName)
   ` const fileContents = fs.readFileSync(fullPath, 'utf8')

    `// Use gray-matter to parse the post metadata section
    `const matterResult = matter(fileContents)

   ` // Combine the data with the id
    `return {
     ` id,
    `  ...matterResult.data
    `}
 ` })
 ` // Sort posts by date
 ` return allPostsData.sort(({ date: a }, { date: b }) => {
  `  if (a < b) {
   `   return 1
 `   } else if (a > b) {
  `    return -1
  `  } else {
    `  return 0
    `}
 ` })
`}

 file. BUT when copy/paste these code as instruction in index.js error will happen
this is my code 

`import_ { getSortedPostsData } from '../lib/posts'`
`_export_ async function getStaticProps() `
`{
  const allPostsData = getSortedPostsData()
  return {
    props: {
      allPostsData
    }
  }
}` 
this is error : Error: ENOENT: no such file or directory, scandir 'C:\Users\user\Desktop\nextjs-blog\posts'

**Comment by icyJoseph** [comment]
Hi,

Do you have a link to a repo with this problem?

I can only think that there's not `posts` folder at the root of the project, did you create it?

**Comment by hermannleboss** [comment]
Change 

`const postsDirectory = path.join(process.cwd(), 'posts')` by `const postsDirectory = path.join(process.cwd(), 'pages/posts');`

**Reply by ughingoro** [reply]
Thank you hermannleboss. This work nicely.

**Comment by ojpro** [comment]
For me, I have moved the posts directory under the `src` directory, like this :
![image](https://github.com/vercel/next.js/assets/108437129/781cb9a0-2800-4047-a254-1c548482c8fe)
and made this change : 
```js 
const postsDirectory = path.join(process.cwd(), 'src/posts');
```

**Reply by SabrioBiblio** [reply]
Does this work in a production environment? I get the error no such file or directory on vercel

**Reply by bilal-tunchie** [reply]
did you find any solution ?
